engines/python/src/main/java/ai/djl/python/engine/PyModel.java

```diff
@@ -304,9 +304,11 @@ private void createAllPyProcesses(int mpiWorkers, int tp) {
             pool = Executors.newFixedThreadPool(mpiWorkers);
         }
         logger.info("Start {} mpiWorkers ...", mpiWorkers);
+        Device device = this.getNDManager().getDevice();
+        int deviceId = device.getDeviceId();
         for (int i = 0; i < mpiWorkers; ++i) {
             logger.debug("Pre-creating python worker: {} ", i);
-            PyProcess worker = new PyProcess(this, pyEnv, i * tp);
+            PyProcess worker = new PyProcess(this, pyEnv, deviceId + (i * tp));
             workerQueue.offer(worker);
```

Comment:
This is for MPI use case. We dont have to take this up now, if it needs thorough testing @frankfliu